modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java

```diff
@@ -382,12 +370,8 @@ protected int getCountInternal(Query query) throws IOException {
             reader.setFilter(filter);
         }
 
-        // setup the target bbox if any, and the generalization hints if available
+        // setup the generalization hints if available
         if (q != null) {
-            if (bbox != null && !bbox.isNull()) {
```

Comment:
This box is used in the new function queryFilterIndex:
https://github.com/geotools/geotools/pull/1056/files#diff-92177fcbab0ca3e01c1a3b7314142838R483

When the filter is a pure spatial filter the code runs https://github.com/geotools/geotools/pull/1056/files#diff-92177fcbab0ca3e01c1a3b7314142838R492 using this box. When the filter mixes alphanumeric and spatial filter the code uses this box when the spatial filter is evaluated.